aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,690 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_sagemaker_model", &resource.Sweeper{
+		Name: "aws_sagemaker_model",
+		F:    testSweepSagemakerModels,
+	})
+}
+
+func testSweepSagemakerModels(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).sagemakerconn
+
+	req := &sagemaker.ListModelsInput{
+		NameContains: aws.String("terraform-testacc-sagemaker-model"),
+	}
+	resp, err := conn.ListModels(req)
+	if err != nil {
+		return fmt.Errorf("Error listing models: %s", err)
+	}
+
+	if len(resp.Models) == 0 {
+		log.Print("[DEBUG] No sagemaker models to sweep")
+		return nil
+	}
+
+	for _, model := range resp.Models {
+		_, err := conn.DeleteModel(&sagemaker.DeleteModelInput{
+			ModelName: model.ModelName,
+		})
+		if err != nil {
+			return fmt.Errorf(
+				"Error deleting sagemaker model (%s): %s",
+				*model.ModelName, err)
+		}
+	}
+
+	return nil
+}
+
+func TestAccAWSSagemakerModel_importBasic(t *testing.T) {
+	rName := acctest.RandString(10)
+	image := "174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"
```

Comment:
Since we use this image in a few places, let's pull it out and make it a constant so we don't have to change in three places if this image ever needs to change

Reply:
Done.